### PR TITLE
Refactor of billing workflow.

### PR DESCRIPTION
**Pull request by wangxf123456**

### CODE REVIEW

File: django_cloud_deploy/cli/prompt.py [hunk 1]
```diff
@@ -19,39 +19,47 @@
...
+            cls, console: io.IO, step_prompt: str, arguments: Dict[str, Any],
+            credentials: credentials.Credentials = None) -> Any:
         """Prompt the user to enter some information.
 
         Args:
             console: Object to use for user I/O.
             step_prompt: A prefix showing the current step number e.g. "[1/3]".
             arguments: The arguments that have already been collected from the
                 user e.g. {"project_id", "project-123"}
+            credentials: (Optional) The OAuth2 Credentials object to use for
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]: How about you remove (Optional) and change the type to Optional[credentials.Credentials]?

File: django_cloud_deploy/cli/prompt.py [hunk 2]
```diff
@@ -445,3 +483,110 @@ def prompt(cls, console: io.IO, step_prompt: str,
...
+    @staticmethod
+    def _get_new_billing_account(
+            console: io.IO,
+            existing_billing_accounts: List[Dict[str, Any]],
+            billing_client: billing.BillingClient) -> str:
+        """Prompt the user for access to the Google credentials.
+
+        Args:
+            console: Object to use for user I/O.
+            existing_billing_accounts:
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]: You missed the description

**Comment by wangxf123456** [comment]
The new workflow is as the follows:
1. Check what billing accounts the user have.
2. Let the user choose one from existing billing accounts, or direct the user to create a new account if the user does not want to use existing accounts.

**Comment by enriquejosepadilla** [comment]
Going to play around with some refactoring regarding this change off of this branch. Please give me until tuesday afternoon for this.

**Comment by enriquejosepadilla** [comment]
> Going to play around with some refactoring regarding this change off of this branch. Please give me until tuesday afternoon for this.

Thanks for the patience :) Went through a rabbit hole of Inheritance and design problem. Went offline with Brian regarding this subject and agreed on current design.

**Pull request merged**